mlem/contrib/kubernetes/base.py

```diff
@@ -0,0 +1,281 @@
+    type: ClassVar = "kubernetes"
+
+    class Config:
+        type_root = True
+
+    def get_host_and_port(
+        self, service, namespace="mlem"  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[str], Optional[int]]:
+        """Returns host and port for the service in Kubernetes"""
+        return None, None
+
+
+class NodePortService(ServiceType):
+
+    type: ClassVar = "k8s_nodeport_service"
+
+    def get_host_and_port(self, service, namespace="mlem"):
+        host, port = None, None
+        port = service.items[0].spec.ports[0].node_port
+        node_name = (
+            client.CoreV1Api()
+            .list_namespaced_pod(namespace)
+            .items[0]
+            .spec.node_name
+        )
+        node_list = client.CoreV1Api().list_node().items
+        node_index = find_index(node_list, node_name)
+        address_dict = node_list[node_index].status.addresses
+        for each_address in address_dict:
+            if each_address.type == "ExternalDNS":
+                host = each_address.address
+        return host, port
+
+
+class LoadBalancerService(ServiceType):
+
+    type: ClassVar = "k8s_loadbalancer_service"
+
+    def get_host_and_port(self, service, namespace="mlem"):
+        host, port = None, None
+        port = service.items[0].spec.ports[0].port
+        ingress = service.items[0].status.load_balancer.ingress[0]
+        host = ingress.hostname or ingress.ip
+        return host, port
+
+
+class K8sDeploymentState(DeployState):
+    """DeployState implementation for Kubernetes deployments"""
+
+    type: ClassVar = "kubernetes"
+
+    image: Optional[DockerImage] = None
+    """Docker Image being used for Deployment"""
+    deployment_name: Optional[str] = None
+    """Name of Deployment"""
+
+
+class K8sDeployment(MlemDeployment, K8sYamlBuildArgs):
+    """MlemDeployment implementation for Kubernetes deployments"""
+
+    type: ClassVar = "kubernetes"
+    state_type: ClassVar = K8sDeploymentState
+    """type of state for Kubernetes deployments"""
+
+    server: Optional[Server] = None
+    """type of Server to use, with options such as FastAPI, RabbitMQ etc."""
+    registry: Optional[DockerRegistry] = DockerRegistry()
+    """docker registry"""
+    daemon: Optional[DockerDaemon] = DockerDaemon(host="")
+    """docker daemon"""
+    kube_config_file_path: Optional[str] = None
+    """path for kube config file of the cluster"""
+    templates_dir: List[str] = []
+    """list of dirs where templates reside"""
+
+    def load_kube_config(self):
+        config.load_kube_config(
+            config_file=self.kube_config_file_path
+            or os.getenv("KUBECONFIG", default="~/.kube/config")
+        )
+
+    def _get_client(self, state: K8sDeploymentState) -> Client:
+        host, port = None, None
+        self.load_kube_config()
+        service = client.CoreV1Api().list_namespaced_service(self.namespace)
+        try:
+            if self.service_type == ServiceTypeEnum.node_port:
```

Comment:
you should make `service_type` a `ServiceType` field and just call `self.service_type.get_host_and_port()`

Reply:
done, used a property instead to calculate the right type